### PR TITLE
Add new contact UI

### LINKED ISSUE #8: Create a new contact UI

### PR DESCRIPTION
**Pull request by mohd-anas-ansari**
Fixes #8

### CODE REVIEW

**Comment by mohd-anas-ansari** [comment]
@karthiknmenon _a

File: app/javascript/src/components/Common/NewPane.jsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+      <div className="px-6">
+        {itemName === "Note" ? (
+          <NewNoteForm onClose={onClose} refetch={fetchNotes} />
+        ) : (
+          <NewContactForm onClose={onClose} refetch={fetchNotes} />
+        )}
+      </div>
+    </Pane>
+  );
+}
```
Comment (karthiknmenon) [inline app/javascript/src/components/Common/NewPane.jsx:26]: Let's refrain from suffixing components or filenames with `Modal, Pane, Form, Page, etc`. Let's use semantic names instead. 
A filename `NewNote` indicates that the file/component handles the logic for creating a new note may it be using a Modal, Pane, or a new page with a form.

File: app/javascript/src/components/Common/TabsHeadersTable/TopHeader.jsx [hunk 2]
```diff
@@ -1,10 +1,10 @@
 import React from "react";
 
-import { Search } from "@bigbinary/neeto-icons";
+import { Search, Plus } from "@bigbinary/neeto-icons";
```
Comment (karthiknmenon) [inline app/javascript/src/components/Common/TabsHeadersTable/TopHeader.jsx:3]: Let's use alias for neeto-icons.

File: app/javascript/src/components/Common/TabsHeadersTable/TopHeader.jsx [hunk 3]
```diff
@@ -1,10 +1,10 @@
...
 import { Header } from "neetoui/layouts";
 import { Input, Button } from "neetoui/v2";
 
-const TopHeader = ({ itemName, setShowNewNotePane }) => {
+const TopHeader = ({ itemName, setShowNewPane }) => {
```
Comment (karthiknmenon) [inline app/javascript/src/components/Common/TabsHeadersTable/TopHeader.jsx:7]: `setShowNewPane` -> `isNewPaneOpen`

File: app/javascript/src/components/Dashboard/Contacts/ContactsTable.jsx [hunk 4]
```diff
@@ -47,7 +55,10 @@ const ContactsTable = ({ contacts }) => {
             dataIndex: "options",
             key: "options",
             render: () => (
-              <MenuHorizontal onClick={() => {}} className="cursor-pointer" />
+              <MenuHorizontal
+                onClick={() => setIsAlertOpen(!isAlertOpen)}
+                className="cursor-pointer"
+              />
```
Comment (karthiknmenon) [inline app/javascript/src/components/Dashboard/Contacts/ContactsTable.jsx:61]: Please move `columnData` to a constant.

File: app/javascript/src/components/Dashboard/Contacts/NewContactForm.jsx [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+  };
+
+  return (
+    <Formik
+      initialValues={{
+        first_name: "",
+        last_name: "",
+        email: "",
+        role: "",
+      }}
```
Comment (karthiknmenon) [inline app/javascript/src/components/Dashboard/Contacts/NewContactForm.jsx:23]: Please move this to a constants file in scope of this directory.

File: app/javascript/src/components/Dashboard/Contacts/NewContactForm.jsx [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+      onSubmit={handleSubmit}
+      validationSchema={yup.object({
+        first_name: yup.string().required("First Name is required"),
+        last_name: yup.string().required("Last Name is required"),
+        email: yup.string().required("Email is required"),
+        role: yup.object().required("Role is required"),
+      })}
```
Comment (karthiknmenon) [inline app/javascript/src/components/Dashboard/Contacts/NewContactForm.jsx:30]: Please move to a constants file.

File: app/javascript/src/components/Dashboard/Contacts/NewContactForm.jsx [hunk 7]
```diff
@@ -0,0 +1,71 @@
+import React from "react";
+
+import { Formik, Form } from "formik";
+import { Toastr, Button } from "neetoui/v2";
+import { Input, Select } from "neetoui/v2/formik";
+import * as yup from "yup";
+
+import { ROLE_DROPDOWN_VALUES } from "constants/contacts";
```
Comment (karthiknmenon) [inline app/javascript/src/components/Dashboard/Contacts/NewContactForm.jsx:8]: Let's keep it in the scope of `Contacts` directory.

File: app/javascript/src/components/Dashboard/Contacts/NewContactForm.jsx [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+
+export default function NewContactForm({ onClose }) {
```
Comment (karthiknmenon) [inline app/javascript/src/components/Dashboard/Contacts/NewContactForm.jsx:10]: Please remove the suffix.

File: app/javascript/src/components/Dashboard/Contacts/NewContactPane.jsx [hunk 9]
```diff
@@ -2,9 +2,9 @@ import React from "react";
 
 import { Pane } from "neetoui";
 
-import NewNoteForm from "./NewNoteForm";
+import NewNoteForm from "./NewContactForm";
 
-export default function NewNotePane({ fetchNotes, showPane, setShowPane }) {
+export default function NewContactPane({ fetchNotes, showPane, setShowPane }) {
```
Comment (karthiknmenon) [inline app/javascript/src/components/Dashboard/Contacts/NewContactPane.jsx:7]: Same here.

**Review by karthiknmenon — COMMENTED** [review]
@anasansari157 _a Please check the suggestions.

**Pull request merged**